app/src/processing/app/Base.java

```diff
@@ -1461,21 +1503,26 @@ public void actionPerformed(ActionEvent actionevent) {
       boardsCustomMenus.add(customMenu);
     }
 
-    List<JMenuItem> menuItemsToClickAfterStartup = new LinkedList<>();
+    menuItemsToClickAfterStartup = new LinkedList<>();
+    boardsButtonGroup = new ButtonGroup();
+    recentBoardsButtonGroup = new ButtonGroup();
+    buttonGroupsMap = new HashMap<>();
 
-    ButtonGroup boardsButtonGroup = new ButtonGroup();
-    Map<String, ButtonGroup> buttonGroupsMap = new HashMap<>();
+    if (BaseNoGui.getRecentlyUsedBoards() != null) {
+      JMenuItem recentLabel = new JMenuItem(tr("Recently used boards"));
+      recentLabel.setEnabled(false);
+      boardMenu.add(recentLabel);
+      rebuildRecentBoardsMenu();
+      //rebuildRecentBoardsMenu(null);
+    }
 
     // Cycle through all packages
-    boolean first = true;
     for (TargetPackage targetPackage : BaseNoGui.packages.values()) {
       // For every package cycle through all platform
       for (TargetPlatform targetPlatform : targetPackage.platforms()) {
 
         // Add a separator from the previous platform
-        if (!first)
-          boardMenu.add(new JSeparator());
-        first = false;
+        boardMenu.add(new JSeparator());
```

Comment:
Seems you fixed this in the next commit by always including the LRU header,  I believe? I guess it will always have a value, since on startup some board will be selected.

Reply:
And a newer commit makes the LRU header again optional (when the preference is set to 0), so I again wonder if that produces a duplicate separator?

Reply:
Love to see that feature...
IMHO the last 4to5 recently used boards are fine, put persistent, not by session, please!
Thank you.